src/Control/Monad/Trans/Either.hs

```diff
@@ -127,3 +132,10 @@ hoistEitherT :: (forall b. m b -> n b) -> EitherT x m a -> EitherT x n a
 hoistEitherT f =
   EitherT . f . runEitherT
 {-# INLINE hoistEitherT #-}
+
+-- | Try an IO action inside an EitherT. If the IO action throws an IOException,
+-- wrap it with the provided handler and return it as a Left inside EitherT.
+tryIOEitherT :: MonadIO m => (IOException -> x) -> IO a -> EitherT x m a
+tryIOEitherT handler =
```

Comment:
I would think that that if this has a callback it should be called `handleXX`, if it just returns `EitherT IOException m a` then `tryXX` would make more sense. Just going along with the conventions in [`Control.Exception`](http://hackage.haskell.org/package/base-4.10.1.0/docs/Control-Exception.html#g:6)

Reply:
So you think it should just be called `tryIO` or `tryEitherT` ?

Reply:
The reason I didn't call it `tryEitherT` was  because it *only* catches `IOException`.

There is is a `tryEitherT` in [x-either](https://github.com/ambiata/x/blob/master/x-eithert/src/X/Control/Monad/Trans/Either.hs#L236) but I'm  not real happy with that because it catches `SomeException` which allows the caller to catch async exceptions which is a bad idea.

Reply:
I guess Im saying I was thinking more like

```
handleIOEitherT :: MonadIO m => (IOException -> x) -> IO a -> EitherT x m a

or 

tryIOEitherT :: MonadIO m => IO a -> EitherT IOException m a

```

which would be more inline with naming conventions in `Control.Exception`

Reply:
If this function was going in `Control.Exception`, I would agree completely.

However, I think the current naming and functionality fits better with the rest of the code in this package. Most of the other functions here are generic in that their return type is `EitherT x m a`.

Reply:
@nhibberd name it`handleIOEitherT` in line with the `handle` function in `Control.Exception`? I can dig that.

The potentially unsafe `tryEitherT` from x-either might be useful with a large documentation warning about bad ideas @erikd if you want to add that here.

Reply:
Sure, I'll go with `handleIOEitherT`.

Reply:
@tmcgilchrist Yeah, might as well add that now, so we only have to do one release :)

Reply:
Winning 👍